assets/src/edit-story/karma/pageAttachment.cuj.karma.js

```diff
@@ -0,0 +1,188 @@
+      await setPageAttachmentLink('http://example.com');
+      await setCtaText('Click me!');
+      const ctaText = fixture.screen.getByText('Click me!');
+      expect(ctaText).toBeDefined();
+    });
+
+    it('it should display warning for a link in the Page Attachment Area', async () => {
+      await addElement();
+      await moveElementToBottom();
+
+      await clickOnTarget(safezone);
+      await setPageAttachmentLink('');
+      const warning = fixture.screen.getByText(
+        'Links can not be located below the dashed line when a page attachment is present. The link to elements found below this line will be removed if you add a page attachment'
+      );
+      expect(warning).toBeDefined();
+    });
+  });
+
+  describe('Action: Remove Page Attachment', () => {
+    it('it should allow removing a Page Attachment', async () => {
+      await setPageAttachmentLink('http://example.com');
+      await setCtaText('Click me!');
+      const ctaText = fixture.screen.getByText('Click me!');
+      expect(ctaText).toBeDefined();
+
+      await setPageAttachmentLink('');
+      expect(fixture.screen.queryByText('Click me!')).toBeNull();
+    });
+  });
+
+  describe('Action: Transforming link with Page Attachment', () => {
+    it('it should display tooltip for a link in Attachment area', async () => {
+      await setPageAttachmentLink('http://example.com');
+      await addElement();
+      await moveElementToBottom(true);
+
+      const popup = fixture.screen.getByText(
+        'Links can not be located below the dashed line when a page attachment is present'
+      );
+      expect(popup).toBeDefined();
+    });
+
+    it('it should cancel link transformation ending in Attachment area', async () => {
+      await setPageAttachmentLink('http://example.com');
+      await addElement();
+      const frameTop = frame.getBoundingClientRect().top;
+      await moveElementToBottom();
+
+      // Verify the same position.
+      expect(frame.getBoundingClientRect().top).toBeDefined(frameTop);
+    });
+  });
+
+  describe('Action: Adding link to element in Attachment area', () => {
+    it('it should not allow adding link to Attachment area', async () => {
+      await setPageAttachmentLink('http://example.com');
+      await addElement(false);
+      await moveElementToBottom();
+
+      const input = fixture.screen.getByLabelText('Edit: Element link');
+      await fixture.events.click(input);
+
+      // Verify that the warning is displayed.
+      const warning = fixture.screen.getByText(
+        'Link can not reside below the dashed line when a page attachment is present'
+      );
+      expect(warning).toBeDefined();
+
+      await fixture.events.keyboard.type('example.com');
+      await input.dispatchEvent(new window.Event('blur'));
+
+      // Verify the link is still null after typing.
+      const {
+        state: {
+          currentPage: {
+            elements: [{ link }],
```

Comment:
Note to self: the first element is actually the background shape, the second element should be tested!